src/components/AmplitudeProvider.js

```diff
@@ -3,62 +3,54 @@ import React from 'react';
 
 import { isValidAmplitudeInstance } from '../lib/validation';
 
-class AmplitudeProvider extends React.Component {
-  componentWillMount() {
-    const { props } = this;
+export const AmplitudeContext = React.createContext({
+  eventProperties: {},
+  instances: {},
+});
 
-    if (isValidAmplitudeInstance(props.amplitudeInstance)) {
-      if (props.apiKey) {
-        props.amplitudeInstance.init(props.apiKey);
+const AmplitudeProvider = ({ amplitudeInstance, apiKey, children, eventProperties, userId }) => {
+  const inheritedContext = React.useContext(AmplitudeContext);
+
+  const contextValue = React.useMemo(() => {
+    return {
+      eventProperties: { ...inheritedContext.eventProperties, ...eventProperties },
+      instances: {
+        ...inheritedContext.instances,
+        [amplitudeInstance._instanceName]: amplitudeInstance,
+      },
+    };
+  }, [
+    amplitudeInstance,
+    amplitudeInstance._instanceName,
+    eventProperties,
+    inheritedContext.eventProperties,
+    inheritedContext.instances,
+  ]);
+
+  React.useEffect(() => {
+    if (isValidAmplitudeInstance(amplitudeInstance)) {
+      if (apiKey) {
+        amplitudeInstance.init(apiKey);
       }
 
-      if (props.userId) {
-        props.amplitudeInstance.setUserId(props.userId);
+      if (userId) {
+        amplitudeInstance.setUserId(userId);
       }
     } else {
       console.error('AmplitudeProvider was not provided with a valid "amplitudeInstance" prop.');
```

Comment:
this warning is not that useful for production environments - it's a validation error and should generally be caught during development. Maybe wrap this in env check (`process.env.NODE_ENV !== 'production'`) to strip it away in production bundles?